src/metabase/pulse.clj

```diff
@@ -261,24 +264,35 @@
          :email (:email recipient)}
         {:kind :user
          :user recipient}))
+    :http
+    []
     (do
      (log/warnf "Unknown channel type %s" (:channel_type pulse-channel))
      [])))
 
-(defn- channel-send!
-  [& args]
-  (try
-    (apply channel/send! args)
-    (catch Exception e
-      ;; Token errors have already been logged and we should not retry.
-      (when-not (and (= :channel/slack (:type (first args)))
-                     (contains? (:errors (ex-data e)) :slack-token))
-        (throw e)))))
-
 (defn- send-retrying!
-  [& args]
+  [pulse-id channel message]
   (try
-    (apply (retry/decorate channel-send!) args)
+    (let [;; once we upgraded to retry 2.x, we can use (.. retry getMetrics getNumberOfTotalCalls) instead of tracking
+          ;; this manually
+          retry-count (volatile! 0)
+          send!       (fn []
+                          (task-history/with-task-history {:task         "channel-send"
+                                                           :task_details {:retry-config   (when retry/*retry-config*
+                                                                                            {:max-attempts (.. ^Retry retry/*retry-config* getRetryConfig getMaxAttempts)})
+                                                                          :channel-type   (:type channel)
+                                                                          :channel-id     (:id channel)
+                                                                          :pulse-id       pulse-id
+                                                                          :retry-attempts @retry-count}}
+                            (try
+                              (channel/send! channel message)
+                              (catch Exception e
+                                (vswap! retry-count inc)
```

Comment:
The meaning of `retry-count` is a little confusing. It actually means the number of failures, not the number of retries. If it fails once and the max-attempts is 1 then `retry-count` at the end will be 1, even though there are zero "retries". So maybe it should be called "failed-attempts"?